bot/core.py

```
@@ -4371,6 +4382,37 @@ def __getattr__(name: str):
+    detail = str(exc).strip() or type(exc).__name__
+    provider = (LLM_PROVIDER_NAME or "").strip().lower()
+    base_url = str(
+        os.getenv("LLM_BASE_URL", "") or os.getenv("OMICSCLAW_BASE_URL", "") or ""
+    ).strip()
+    if not base_url:
+        try:
+            from omicsclaw.core.provider_runtime import get_active_provider_runtime
+
+            runtime = get_active_provider_runtime()
+            base_url = str(getattr(runtime, "base_url", "") or "").strip()
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)